concourse/pipelines/replicator.py

```diff
@@ -385,16 +386,19 @@ def process_results(self, results):
         )
 
         all_notifications_succeeded = True
-        for failed_descriptor in failed_descriptors:
-            warning(failed_descriptor.definition_descriptor.pipeline_name)
-            try:
-                self._notify_broken_definition_owners(failed_descriptor)
-            except Exception:
-                warning('an error occurred whilst trying to send error notifications')
-                traceback.print_exc()
-                all_notifications_succeeded = False
-
-        # signall error only if error notifications failed
+        if self._notify_owners:
```

Comment:
mv the notification coding into a method
call said method iff it is configured (as of the notify_owners attribute)